logstash-core/src/main/java/org/logstash/config/ir/PipelineIR.java

```diff
@@ -51,7 +51,7 @@ public PipelineIR(Graph inputSection, Graph filterSection, Graph outputSection,
 
         this.graph.validate();
 
-        if (this.getOriginalSource() != null && this.getOriginalSource().matches("^\\S+$")) {
+        if (this.getOriginalSource() != null && !this.getOriginalSource().matches("^\\s+$")) {
```

Comment:
@andrewvc btw. I have no idea how often this is called on average, but this can be way optimized by using a compiled regex.
Or better yet ... to avoid creating the `Matcher` instance just using:

```java
    private static boolean spacesOnly(final CharSequence chars) {
        for (int i = 0; i < chars.length(); ++i) {
            if (chars.charAt(i) != ' ') {
                return false;
            }
        }
        return true;
    }
```